core/history_api.php

```diff
@@ -510,6 +510,203 @@ function history_get_raw_events_array( $p_bug_id, $p_user_id = null, $p_start_ti
 }
 
 /**
+ * Localize the specified field name for native or custom fields.
+ *
+ * @param string $p_field_name The field name.
+ * @return string The localized field name.
+ */
+function history_localize_field_name( $p_field_name ) {
+	switch( $p_field_name ) {
+		case 'category':
+			$t_field_localized = lang_get( 'category' );
+			break;
+		case 'status':
+			$t_field_localized = lang_get( 'status' );
+			break;
+		case 'severity':
+			$t_field_localized = lang_get( 'severity' );
+			break;
+		case 'reproducibility':
+			$t_field_localized = lang_get( 'reproducibility' );
+			break;
+		case 'resolution':
+			$t_field_localized = lang_get( 'resolution' );
+			break;
+		case 'priority':
+			$t_field_localized = lang_get( 'priority' );
+			break;
+		case 'eta':
+			$t_field_localized = lang_get( 'eta' );
+			break;
+		case 'view_state':
+			$t_field_localized = lang_get( 'view_status' );
+			break;
+		case 'projection':
+			$t_field_localized = lang_get( 'projection' );
+			break;
+		case 'sticky':
+			$t_field_localized = lang_get( 'sticky_issue' );
+			break;
+		case 'project_id':
+			$t_field_localized = lang_get( 'email_project' );
+			break;
+		case 'handler_id':
+			$t_field_localized = lang_get( 'assigned_to' );
+			break;
+		case 'reporter_id':
+			$t_field_localized = lang_get( 'reporter' );
+			break;
+		case 'version':
+			$t_field_localized = lang_get( 'product_version' );
+			break;
+		case 'fixed_in_version':
+			$t_field_localized = lang_get( 'fixed_in_version' );
+			break;
+		case 'target_version':
+			$t_field_localized = lang_get( 'target_version' );
+			break;
+		case 'date_submitted':
+			$t_field_localized = lang_get( 'date_submitted' );
+			break;
+		case 'last_updated':
+			$t_field_localized = lang_get( 'last_update' );
+			break;
+		case 'os':
+			$t_field_localized = lang_get( 'os' );
+			break;
+		case 'os_build':
+			$t_field_localized = lang_get( 'os_version' );
+			break;
+		case 'build':
+			$t_field_localized = lang_get( 'build' );
+			break;
+		case 'platform':
+			$t_field_localized = lang_get( 'platform' );
+			break;
+		case 'summary':
+			$t_field_localized = lang_get( 'summary' );
+			break;
+		case 'duplicate_id':
+			$t_field_localized = lang_get( 'duplicate_id' );
+			break;
+		case 'sponsorship_total':
+			$t_field_localized = lang_get( 'sponsorship_total' );
+			break;
+		case 'due_date':
+			$t_field_localized = lang_get( 'due_date' );
+			break;
+		default:
```

Comment:
What about
```php
		case 'category':
		case 'status':
		case 'severity':
		case 'reproducibility':
		case 'resolution':
		case 'priority':
		case 'eta':
		case 'projection':
		case 'fixed_in_version':
		case 'target_version':
		case 'date_submitted':
		case 'os':
		case 'build':
		case 'platform':
		case 'summary':
		case 'duplicate_id':
		case 'sponsorship_total':
		case 'due_date':
			$t_field_localized = lang_get( $p_field_name );
			break;
		case 'view_state':
			$t_field_localized = lang_get( 'view_status' );
			break;
		case 'sticky':
			$t_field_localized = lang_get( 'sticky_issue' );
			break;
		case 'project_id':
			$t_field_localized = lang_get( 'email_project' );
			break;
		case 'handler_id':
			$t_field_localized = lang_get( 'assigned_to' );
			break;
		case 'reporter_id':
			$t_field_localized = lang_get( 'reporter' );
			break;
		case 'version':
			$t_field_localized = lang_get( 'product_version' );
			break;
		case 'last_updated':
			$t_field_localized = lang_get( 'last_update' );
			break;
		case 'os_build':
			$t_field_localized = lang_get( 'os_version' );
			break;
```

Reply:
Same comment.  This current code is more friendly to moving to constants.

Reply:
I thought that having the field names as some kind of unique index at various places is quite good, e.g.

- no need for additional constants
- adding new fields is easy, no need to add code at various places
- short and simple code
- better code coverage, even when doing just a few tests

After some refactoring by eliminating historic inconsistencies (e.g. `view_state` vs `view_status`) there would be some more advantages

- complete removal of the switch cases
- better performance, especially as PHP is quite bad in terms of how `switch` statements are executed internally

Reply:
I'm not sure I understand the proposal here, but I agree we need some cleanup and we should create a code cleanup issue to discuss.  It may be that you are proposing having a structure like cfdefs for native fields that maps a field to all associated constants and behaviors.  In fact, I think it would be useful to see if we can have a common structure for all fields that is common cross for native and custom fields.